src/dotnet/commands/dotnet-build3/ForwardingApp.cs

```diff
@@ -0,0 +1,106 @@
+    {
+        private static readonly string s_hostExe = "dotnet";
+
+        private string _forwardApplicationPath;
+        private string[] _argsToForward;
+        private string _depsFile;
+        private string _runtimeConfig;
+        private string _additionalProbingPath;
+        private Dictionary<string, string> _environment;
+
+        private string[] _allArgs;
+        
+        public ForwardingApp(
+            string forwardApplicationPath, 
+            string[] argsToForward, 
+            string depsFile = null, 
+            string runtimeConfig = null,
+            string additionalProbingPath = null,
+            Dictionary<string, string> environment = null)
+        {
+            _forwardApplicationPath = forwardApplicationPath;
+            _argsToForward = argsToForward;
+            _depsFile = depsFile;
+            _runtimeConfig = runtimeConfig;
+            _additionalProbingPath = additionalProbingPath;
+            _environment = environment;
+
+            _allArgs = new string[] { _forwardApplicationPath }.Concat(_argsToForward).ToArray();
+
+            if (_depsFile != null)
+            {
+                _allArgs = new string[] { "--depsfile", _depsFile }.Concat(_allArgs).ToArray();
+            }
+
+            if (_runtimeConfig != null)
+            {
+                _allArgs = new string[] { "--runtimeconfig", _runtimeConfig }.Concat(_allArgs).ToArray();
+            }
+
+            if (_additionalProbingPath != null)
+            {
+                _allArgs = new string[] { "--additionalprobingpath", _additionalProbingPath }.Concat(_allArgs).ToArray();
+            }
+
+            _allArgs = new string[] { "exec" }.Concat(_allArgs).ToArray();
+
+            Console.WriteLine(_forwardApplicationPath);
+            Console.WriteLine(_argsToForward);
+            Console.WriteLine(_depsFile);
+            Console.WriteLine(_runtimeConfig);
+            Console.WriteLine(_additionalProbingPath);
+            Console.WriteLine(ArgumentEscaper.EscapeAndConcatenateArgArrayForProcessStart(_allArgs));
+        }
+
+        public int Execute()
+        {
+            var processInfo = new ProcessStartInfo()
+            {
+                FileName = GetHostExeName(),
+                Arguments = ArgumentEscaper.EscapeAndConcatenateArgArrayForProcessStart(_allArgs),
+                UseShellExecute = false
+            };
+
+            if (_environment != null)
+            {
+                foreach (var entry in _environment)
+                {
+                    processInfo.Environment[entry.Key] = entry.Value;
+                }
+            }
+
+            var process = new Process()
+            {
+                StartInfo = processInfo
+            };
+
+            process.Start();
+            process.WaitForExit();
+
+            return process.ExitCode;
+        }
+
+        private string GetHostExeName()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
It would be nice to reuse some existing logic here.  How about `Microsoft.DotNet.ProjectModel.FileNameSuffixes.CurrentPlatform.Exe"?